### PR TITLE
Add pawn promotion detection

### PR DESCRIPTION
**Pull request by swcraig**
`MoveType` recognises possible pawn promotion moves. These are very similar to `RegularMove`s but the game flow will need to notify the user, who will be prompted to choose which piece type the pawn will become.

### CODE REVIEW

File: lib/move_type.rb [hunk 1]
```diff
@@ -23,6 +28,11 @@ def pawn_move_type(move)
       end
     end
 
+    def possible_pawn_promotion?(move)
+      [PLAYER_ONE_STARTING_RANK, PLAYER_TWO_STARTING_RANK].
+        include?(move.destination_cell.rank)
```
Comment (gmacdougall) [inline lib/move_type.rb]: You can identify a major code smell here in that this is a utility function (https://github.com/troessner/reek/blob/master/docs/Utility-Function.md). It doesn't refer to instance state in any way

Additionally, this method takes in one parameter that's an object. Why does this method belong here and not on `move`?

When I'm thinking about code. I think about the code that I want to be able to write.

If I were to ask myself when does a pawn get promoted? My answer would be "when it gets to the last row".

The code I want to be able to write is:

```Ruby
move.to_last_row?
```

Comment (swcraig) [inline lib/move_type.rb]: Thank you for the feedback Gregor. I have never thought about something like this.

File: lib/player_requested_move.rb [hunk 2]
```diff
@@ -33,6 +35,11 @@ def making_forward_move?(movement_vector)
       (movement_vector.norm * FORWARD_MOVE.norm).abs
   end
 
+  def to_last_rank?
+    [PLAYER_ONE_STARTING_RANK, PLAYER_TWO_STARTING_RANK].
+      include?(@destination_cell.rank)
```
Comment (gmacdougall) [inline lib/player_requested_move.rb]: We can take this refactor a step further. What deserves the responsibility of determining if a Cell is in the last row of the board?

I think that `Board` is the place that comes to mind first, though `Cell` is also a possibility. I don't think `MoveType` is the correct answer.

I would have this be:

```Ruby
def to_last_rank?
  @destination_cell.last_rank?
end
```

File: spec/move_type_spec.rb [hunk 3]
```diff
@@ -34,5 +36,17 @@
     context 'when it is a regular move' do
       it { is_expected.to be_a RegularMove }
     end
+
+    context "when a pawn is moving into the opponent's starting rank" do
+      let(:source) { cell_e7 }
+      let(:destination) { cell_e8 }
+
+      before do
+        board.cell_at(cell_e7).content = board.cell_at(cell_e2).content
```
Comment (jacquesporveau) [inline spec/move_type_spec.rb]: I believe what this is doing is moving the pawn to the second last last row and clear the opponents pieces. Maybe I am wrong but I am not sure. Maybe we can add a single line comment to tell ppl what this is doing from a high level.

**Pull request merged**